extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -715,15 +707,26 @@ export class Bind {
    * @private
    */
   scanAttribute_(attribute, element) {
-    const {tagName} = element;
-    const {name} = attribute;
-    if (name.length > 2 && name[0] === '[' && name[name.length - 1] === ']') {
-      const property = name.substr(1, name.length - 2);
-      if (this.validator_.canBind(tagName, property)) {
+    const tag = element.tagName;
+    const attr = attribute.name;
+
+    let property;
+    if (attr.length > 2 && attr[0] === '[' && attr[attr.length - 1] === ']') {
+      property = attr.substr(1, attr.length - 2);
+    } else if (startsWith(attr, 'data-amp-bind-')) {
```

Comment:
interesting, I initially thought is is allowing any `data-amp-bind-*` to become bindable via `[data-amp-bind-*]` ( I actually ran into a case few days ago that I wanted to use a `data-` attribute in CSS instead ) but I see that this is an alternative to `[foo]`. 

Given we may want to allow arbitrary `data-` attributes to become bindable in the feature (like how we do with `aria-*`), I wonder if we should remove `data-` prefix from this one and make it `amp-bind-*` and add a general validation/sanitizer rule for it.

Reply:
Updated the PR description to make this more clear.

It's still safe for binding to `data-*` attributes since the expected prefix is the more specific `data-amp-bind-*`. I don't think we'll ever want to support something like `[data-amp-bind-foo]`.

You're right that the validator won't enforce rules for `data-amp-bind-*`, but I think this is fine for a "workaround" syntax like this one. It'll also avoid yet another source of skew (between validator rules for `[foo]` vs. `data-amp-bind-foo` vs. `bind-validator.js`).